src/app/services/public-comment.service.ts

```diff
@@ -0,0 +1,111 @@
+import { Injectable } from '@angular/core';
+import { Response } from '@angular/http';
+import { PublicComment } from '../models/publiccomment';
+import { Comment } from '../models/comment';
+import { Document } from '../models/document';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+
+import { Api } from './api';
+import { ValuedComponent } from '../models/vcs';
+
+@Injectable()
+export class PublicCommentService {
+  pcp: PublicComment;
+
+  constructor(private api: Api) { }
+
+  getByCode(id: string, code: string): Observable<PublicComment> {
+    this.pcp = null;
+
+    // Grab the project data first
+    return this.api.getPCPByCode(id)
+      .map((res: Response) => res.json())
+      .map((pcp: PublicComment) => {
+        if (!pcp) {
+          return Observable.throw(new Error('PCP not found'));
+        }
+
+        this.pcp = pcp;
+        this.pcp.relatedDocuments.forEach((document, index ) => {
+          document = new Document(document);
+          this.processDocuments(this.pcp.relatedDocuments, document, index);
+        });
+        this.setStatus(new Date(this.pcp.dateStarted), new Date(this.pcp.dateCompleted));
+        this.pcp.comments = Array<Comment>();
+        this.pcp.vcs = [];
+        return this.pcp;
+      })
+      .switchMap(() => this.getCommentsByPCP(id))
+      .switchMap(() => this.getCommentVcs())
+      .switchMap(() => this.getProjectByCode(code))
```

Comment:
@btzc if the order of this calls matters, it may be worth putting a comment somewhere explaining why

Reply:
Fixed